crypto/keys/keybase.go

```diff
@@ -388,8 +392,15 @@ func (kb dbKeybase) Delete(name, passphrase string, skipPass bool) error {
 		}
 	}
 
-	kb.db.DeleteSync(addrKey(info.GetAddress()))
-	kb.db.DeleteSync(infoKey(name))
+	err = kb.db.DeleteSync(addrKey(info.GetAddress()))
+	if err != nil {
+		return err
+	}
+
+	err = kb.db.DeleteSync(infoKey(name))
```

Comment:
If the first call (`kb.db.DeleteSync(addrKey(info.GetAddress()))`) succeeds and this (`kb.db.DeleteSync(infoKey(name))`) fails, we'd end up with a key reference stored by name yet with no way to access it via address (note: this problem is not introduced by this change).
My suggestion is that if the second Delete() call fails, we then try to reintroduce the reference by name too.

Reply:
I don't want to block this PR because of this. This is good work towards the right direction, maybe @marbar3778 open a new issue that we will tackle in a separate PR eventually?

Reply:
https://github.com/cosmos/cosmos-sdk/issues/5841

Reply:
There is no need to create an issue or even block on this. It's super simple:

```go
	batch := kb.db.NewBatch()
	defer batch.Close()

	batch.Delete(addrKey(info.GetAddress()))
	batch.Delete(infoKey(name))

	if err := batch.WriteSync(); err != nil {
		return err
	}
```